### PR TITLE
[C-862] Add env to saga context, fix eagerDiscprov in native

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Adds env to saga context, making it easier to work with process.env and Config
Fixes eagerDiscprov in native, which was using window.localStorage before

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-862-add-env-to-saga-context

File: packages/mobile/src/services/env.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const env = {
+  EAGER_DISCOVERY_NODES: process.env.REACT_APP_EAGER_DISCOVERY_NODES
+}
```
Comment (sliptype) [inline packages/mobile/src/services/env.ts:3]: It probably makes sense to refactor audiusBackend to use this too. But can do that later

File: packages/web/src/common/services/audius-api-client/AudiusAPIClient.ts [hunk 2]
```diff
@@ -1421,7 +1429,9 @@ export class AudiusAPIClient {
     }
 
     // Set the state to the eager discprov
-    const eagerDiscprov = getEagerDiscprov()
+    console.log('aight its getting eager')
```
Comment (sliptype) [inline packages/web/src/common/services/audius-api-client/AudiusAPIClient.ts:1432]: Logs

**Review by sliptype — APPROVED** [review]
Looks good!

**Pull request merged**